Screenshot Maker.cpp

```diff
@@ -38,36 +39,66 @@ int APIENTRY wWinMain(HINSTANCE hInstance,
 
 	bool defaultfn = true;
 	const wchar_t* filename = L"screenshot.png";
+	wchar_t format[5] = L"png";
 	wchar_t encoder[16] = L"image/png";
 	long quality = -1;
 	int resize = -1;
 
-	for(int i=1; i<__argc-1; i++)
+#define ARGV(x) wcscmp(__wargv[i], x) == 0
+	for(int i=1; i<__argc; i++)
```

Comment:
By looping until `__argc` instead of `__argc-1`, there's a potentially out-of-bounds access, e.g. for `screenshot_maker.exe -encoder`.